### DISCUSSION TITLE
Recommendation for execution flow to ensure atomic registration of user in Ory and backend

### DISCUSSION THREAD

**Discussion by Sakdow** [ory/kratos#3006] [Q&A]
Hi there,

I am building a web application with a Svelte frontend and a Go backend, and I want to use Ory Cloud to manage user authentication and authorization. According to this [video](https://www.youtube.com/watch?v=umSRQBVdPjM), the frontend initiates a request to Ory to initialize a registration flow, then gets a `session_token` from the response, and then calls the backend to retrieve the user information that has been created on Ory (such as the email) and save it in our own database for business needs.

I need this process to be "atomic", i.e., I do not want the frontend to make two calls to avoid the rare case where the user is present on Ory but not in our database. Instead, I need Ory and the backend (in case of success) to each register the user after a single call. Can you recommend an execution flow that addresses these requirements? Alternatively, if I'm completely wrong, let me know what the best way is to maintain consistency in my data.

Thanks in advance for your help.

Best regards,
Nathan Roche

**Comment by vinckr** [comment]
Hello Roche, 
I think a `after registration` [webhook](https://www.ory.sh/docs/kratos/hooks/configure-hooks) to your backend would be a good solution.
